Dockerfile

```diff
@@ -0,0 +1,33 @@
+# Build the manager binary
+FROM golang:1.23 AS builder
```

Comment:
Can we try ans see if it will work with
FROM registry.access.redhat.com/ubi9/go-toolset:1.21 AS builder

?

This will keep us aligned later down the road when we want to build product containers, and that one is openly available to use too.

Downside: it's an older golang compiler

Reply:
Testing it locally.... let's see...

Reply:
Nah, but it's a different discussion all together. We may want to try and use a lower version of golang that it's available downstream like 1.22

Reply:
registry.access.redhat.com/ubi9/go-toolset:1.22.9 seems to compile everything fine if we downgrade the go.mod reqs to 1.22.9

Reply:
I prefer that we stay on the safe side... if we introduce something depending on go 1.23 we fail until the downstream also has that.

Reply:
Failing for me locally:

```
[1/2] STEP 11/11: RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
command-line-arguments: go build command-line-arguments: copying /tmp/go-build2523334224/b001/exe/a.out: open manager: permission denied
```

Is `/tmp` somehow different on ubi?

Reply:
So the default user on ubi is `default`, but the build directory `/workspace` is owned by root.